File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/monitoring/port/RagQualityMetricsPort.java
package com.study.webflux.rag.application.monitoring.port;

public interface RagQualityMetricsPort {
	void recordMemoryCandidateCount(int count);

	void recordMemoryFilteredCount(int count);

	void recordMemorySimilarityScore(double score);

	void recordMemoryImportanceScore(double importance);

	void recordDocumentRelevanceScore(double score);
}

File: backend/monitoring/src/main/kotlin/com/miyou/app/monitoring/port/RagQualityMetricsPort.kt
package com.miyou.app.monitoring.port

interface RagQualityMetricsPort {
    fun recordMemoryCandidateCount(count: Int)

    fun recordMemoryFilteredCount(count: Int)

    fun recordMemorySimilarityScore(score: Double)

    fun recordMemoryImportanceScore(importance: Double)

    fun recordDocumentRelevanceScore(score: Double)
}
